src/java/org/apache/cassandra/config/BitRate.java

```diff
@@ -0,0 +1,331 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.config;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.google.common.primitives.Ints;
+
+/**
+ * Represents a bit rate type used for cassandra configuration. It supports the opportunity for the users to be able to
+ * add units to the confiuration parameter value. (CASSANDRA-15234)
+ */
+public final class BitRate
+{
+    /**
+     * The Regexp used to parse the rate provided as String in cassandra.yaml.
+     */
+    private static final Pattern BIT_RATE_UNITS_PATTERN = Pattern.compile("^(\\d+)(MiB/s|mib/s|MiB/S|KiB/s|kib/s|KiB/S|B/s|b/s|B/S)$");
+
+    private final long quantity;
+
+    private final BitRateUnit unit;
+
+    public BitRate(String value)
+    {
+        //parse the string field value
+        Matcher matcher = BIT_RATE_UNITS_PATTERN.matcher(value);
+
+        if (!matcher.find())
+        {
+            throw new IllegalArgumentException("Invalid bit rate: " + value + " Accepted units: MiB/s, KiB/s, B/s.");
+        }
+
+        quantity = Long.parseLong(matcher.group(1));
+        unit = BitRateUnit.fromSymbol(matcher.group(2));
+    }
+
+    private BitRate(long quantity, BitRateUnit unit)
+    {
+        this.quantity = quantity;
+        this.unit = unit;
+    }
+
+    /**
+     * Creates a {@code BitRate} of the specified amount of bits per second.
+     *
+     * @param bitsPerSeconds the amount of bits per second
+     * @return a {@code BitRate}
+     */
+    public static BitRate inBitsPerSeconds(long bitsPerSeconds)
+    {
+        return new BitRate(bitsPerSeconds, BitRateUnit.BITS_PER_SECOND);
+    }
+
+    /**
+     * Creates a {@code BitRate} of the specified amount of kilobits per second.
+     *
+     * @param kilobitsPerSeconds the amount of kilobits per second
+     * @return a {@code BitRate}
+     */
+    public static BitRate inKilobitsPerSecond(long kilobitsPerSeconds)
+    {
+        return new BitRate(kilobitsPerSeconds, BitRateUnit.KILOBITS_PER_SECOND);
+    }
+
+    /**
+     * Creates a {@code BitRate} of the specified amount of megabits per second.
+     *
+     * @param megabitsPerSeconds the amount of megabits per second
+     * @return a {@code BitRate}
+     */
+    public static BitRate inMegabitsPerSecond(long megabitsPerSeconds)
+    {
+        return new BitRate(megabitsPerSeconds, BitRateUnit.MEGABITS_PER_SECOND);
+    }
+
+    /**
+     * Returns the bit rate unit.
+     *
+     * @return the bit rate unit.
```

Comment:
nit/style: Given the audience is mostly people reading the code anyway, it's probably okay to just have the `@return` info here and not duplicate essentially the same thing in the general method description. (Not binding at all.)

Reply:
I agree with you, I was told to be adding more javadoc and I think I overdid it. :)

Reply:
I will remove similar other occurrences too. Revising now. Thanks